recipes/odbc/all/conanfile.py

```diff
@@ -27,84 +29,106 @@ class OdbcConan(ConanFile):
         "with_libiconv": True,
     }
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     @property
     def _user_info_build(self):
         return getattr(self, "user_info_build", self.deps_user_info)
 
     def export_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+        try:
+            del self.settings.compiler.cppstd
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.libcxx
+        except Exception:
+            pass
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def requirements(self):
+        self.requires("libtool/2.4.7")
         if self.options.with_libiconv:
             self.requires("libiconv/1.17")
 
     def validate(self):
-        if self.settings.os == "Windows":
+        if self.info.settings.os == "Windows":
             raise ConanInvalidConfiguration("odbc is a system lib on Windows")
 
     def build_requirements(self):
-        self.build_requires("gnu-config/cci.20201022")
+        self.tool_requires("gnu-config/cci.20210814")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        yes_no = lambda v: "yes" if v else "no"
+        tc.configure_args.extend([
+            "--without-included-ltdl",
+            f"--with-ltdl-include={self.dependencies['libtool'].cpp_info.includedirs[0]}",
+            f"--with-ltdl-lib={self.dependencies['libtool'].cpp_info.libdirs[0]}",
+            "--disable-ltdl-install",
+            f"--enable-iconv={yes_no(self.options.with_libiconv)}",
+            "--sysconfdir=/etc",
+        ])
+        if self.options.with_libiconv:
+            libiconv_prefix = self.dependencies["libiconv"].package_folder
+            tc.configure_args.append(f"--with-libiconv-prefix={libiconv_prefix}")
+        tc.generate()
 
     def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+        apply_conandata_patches(self)
+        # support more triplets
         shutil.copy(self._user_info_build["gnu-config"].CONFIG_SUB,
```

Comment:
Files copy?

Reply:
How?

Reply:
The new `tools.copy` it does not so dummy like `self.copy`:
https://docs.conan.io/en/latest/reference/conanfile/tools/files/basic.html#conan-tools-files-copy

Reply:
Ideal the Conan api should be doing file io `tools.files` api